File: phonegap/ios/git/www/js/triangle.js
function Triangle(rx,  ry,  rz,  px1,  py1,  px2,  py2,  px3,  py3,  pr, color)  {
	this.partinit();
	this.setRoot(rx, ry, rz, pr);
	this.setColor(color);
	this.x1 = px1;
	this.y1 = py1;
	this.x2 = px2;
	this.y2 = py2;
	this.x3 = px3;
	this.y3 = py3;   
}		

Triangle.prototype= new Part();
	
Triangle.prototype.reset= function() {
	this.partreset();
};

Triangle.prototype.getNumberOfData= function() {
		return 4+2 * (3+1); // type, n coord, color, data 3*(x & y)
};
	
Triangle.prototype.getData= function(d, startD,  xn,  yn,  zn,  a11,  a21,  a12,  a22) {
		var n= startD;
		d[n++]= this.getType();
		d[n++]= 3;
		d[n++] =  (this.a<<24)+(this.r<<16)+(this.g<<8)+this.b;
		d[n++] = 1;
		n+=2;
		var phi = (this.rot + this.rrot);
		phi= Math.round(phi);
		while (phi < 0) {
			phi += 360;
		}
		phi %= 360;
		var sinbeta = mysin[phi];
		var cosbeta = mycos[phi];
		var dummy;
		var dummy2;
		dummy = this.x1 * cosbeta - this.y1 * sinbeta +this.rx +this.x;
		dummy2 = this.x1 * sinbeta + this.y1 * cosbeta+ this.ry+ this.y;
		d[n] = Math.round(dummy * a11 + dummy2 * a12 + xn);
		d[n+1] = Math.round(dummy * a21 + dummy2 * a22 + yn);
		dummy = this.x2 * cosbeta - this.y2 * sinbeta +this.rx +this.x;
		dummy2 = this.x2 * sinbeta + this.y2 * cosbeta+ this.ry+ this.y;
		d[n+2] = Math.round(dummy * a11 + dummy2 * a12 + xn);
		d[n + 3] = Math.round(dummy * a21 + dummy2 * a22 + yn);
		dummy = this.x3 * cosbeta - this.y3 * sinbeta +this.rx +this.x;
		dummy2 = this.x3 * sinbeta + this.y3 * cosbeta+ this.ry+ this.y;
		d[n + 4] = Math.round(dummy * a11 + dummy2 * a12 + xn);
		d[n + 5] = Math.round(dummy * a21 + dummy2 * a22 + yn);
		return this.getNumberOfData();
};

Triangle.prototype.getType= function() {
		return TRIANGLE;
};

File: java/src/main/java/de/digitalemil/eagle/Triangle.java
package de.digitalemil.eagle;

public class Triangle extends Part {
	protected float x1, y1, x2, y2, x3, y3;

	public Triangle(float rx, float ry, float rz, float px1, float py1,
			float px2, float py2, float px3, float py3, float pr, int color) {
		setRoot(rx, ry, rz, pr);
		setColor(color);
		x1 = px1;
		y1 = py1;
		x2 = px2;
		y2 = py2;
		x3 = px3;
		y3 = py3;
	}

	public int getNumberOfData() {
		return 4 + 2 * (3 + 1); // type, n coord, color, data 3*(x & y)
	};

	public int getData(float[] d, int startD, float xn, float yn, float zn,
			float a11, float a21, float a12, float a22) {
		int n = startD;
		d[n++] = getType();
		d[n++] = 3;
		d[n++] = (a << 24) + (r << 16) + (g << 8) + b;
		d[n++] = 1;
		n += 2;
		int phi = calcPhi(rot+ rrot);
		float sinbeta = mysin[phi];
		float cosbeta = mycos[phi];
		float dummy;
		float dummy2;
		if (this.coordtap != null) {
			float rx1 = this.rx + this.x;
			float ry1 = this.ry + this.y;
			this.coordtap.save(Math.round(rx1 * a11 + ry1 * a12 + xn), Math.round(rx1 * a21 + ry1 * a22 + yn), 0, 0, 0, 0, 0);
		}
		dummy = x1 * cosbeta - y1 * sinbeta + rx + x;
		dummy2 = x1 * sinbeta + y1 * cosbeta + ry + y;
		d[n] = Math.round(dummy * a11 + dummy2 * a12 + xn);
		d[n + 1] = Math.round(dummy * a21 + dummy2 * a22 + yn);
		dummy = x2 * cosbeta - y2 * sinbeta + rx + x;
		dummy2 = x2 * sinbeta + y2 * cosbeta + ry + y;
		d[n + 2] = Math.round(dummy * a11 + dummy2 * a12 + xn);
		d[n + 3] = Math.round(dummy * a21 + dummy2 * a22 + yn);
		dummy = x3 * cosbeta - y3 * sinbeta + rx + x;
		dummy2 = x3 * sinbeta + y3 * cosbeta + ry + y;
		d[n + 4] = Math.round(dummy * a11 + dummy2 * a12 + xn);
		d[n + 5] = Math.round(dummy * a21 + dummy2 * a22 + yn);
		return getNumberOfData();
	};

	public int getType() {
		return Types.TRIANGLE;
	}
}
